code/modules/hydroponics/hydroponics.dm

```diff
@@ -454,13 +452,15 @@
 		harvest = FALSE
 		weedlevel = 0 // Reset
 
-		sleep(5) // Wait a while
-		update_appearance()
-		visible_message("<span class='warning'>The mutated weeds in [src] spawn some [myseed.plantname]!</span>")
-		TRAY_NAME_UPDATE
+		var/message = "<span class='warning'>The mutated weeds in [src] spawn some [myseed.plantname]!</span>"
+		addtimer(CALLBACK(src, .proc/after_wait, message), 0.5 SECONDS)
 	else
 		to_chat(usr, "<span class='warning'>The few weeds in [src] seem to react, but only for a moment...</span>")
 
+/obj/machinery/hydroponics/proc/after_wait(message)
+		update_appearance()
+		visible_message(message)
+		TRAY_NAME_UPDATE
```

Comment:
probably needs documentation for this proc and a better name

Reply:
i'll document it, but i can't think of a good name to give to it

Reply:
maybe like "after_mutation"